HTTP_API.md

```diff
@@ -0,0 +1,952 @@
+| **Recordings in target JVMs**                                             |                                                                         |
+| Get a list of recordings in a target JVM                                  | [`TargetRecordingsGetHandler`](#TargetRecordingsGetHandler)             |
+| Create a snapshot recording in a target JVM                               | [`TargetSnapshotPostHandler`](#TargetSnapshotPostHandler)               |
+| Start a recording in a target JVM                                         | [`TargetRecordingsPostHandler`](#TargetRecordingsPostHandler)           |
+| Stop a recording in a target JVM                                          | [`TargetRecordingPatchHandler`](#TargetRecordingPatchHandler)           |
+| Delete a recording in a target JVM                                        | [`TargetRecordingDeleteHandler`](#TargetRecordingDeleteHandler)         |
+| Download a recording in a target JVM                                      | [`TargetRecordingGetHandler`](#TargetRecordingGetHandler)               |
+| Download a report of a recording in a target JVM                          | [`TargetReportGetHandler`](#TargetReportGetHandler)                     |
+| Save a recording in a target JVM to persistent storage                    | [`TargetRecordingPatchHandler`](#TargetRecordingPatchHandler)           |
+| Upload a recording from a target JVM to the Grafana datasource            | [`TargetRecordingUploadPostHandler`](#TargetRecordingUploadPostHandler) |
+| **Recordings in persistent storage**                                      |                                                                         |
+| Get a list of recordings in persistent storage                            | [`RecordingsGetHandler`](#RecordingsGetHandler)                         |
+| Upload a recording to persistent storage                                  | [`RecordingsPostHandler`](#RecordingsPostHandler)                       |
+| Delete a recording from persistent storage                                | [`RecordingDeleteHandler`](#RecordingDeleteHandler)                     |
+| Download a recording in persistent storage                                | [`RecordingGetHandler`](#RecordingGetHandler)                           |
+| Download a report of a recording in persistent storage                    | [`ReportGetHandler`](#ReportGetHandler)                                 |
+| Upload a recording from persistent storage to the Grafana datasource      | [`RecordingUploadPostHandler`](#RecordingUploadPostHandler)             |
+
+
+### Core
+
+* #### `AuthPostHandler`
+
+    ###### synopsis
+    Attempts user authenticaiton;
+    used as a simple way to check a user authentication header.
+
+    ###### request
+    `POST /api/v1/auth`
+
+    The request should include an `Authorization` header to be checked.
```

Comment:
Should this explain the format of the `Authorization` header, ex. `Basic` vs `Bearer`? I think it's also worth including a short explanation of how the different `AuthManager`s influence the interpretation of this header - `README.md` goes into some detail on it already so the text here can be shorter, or even just refer back to the README.